Android/Narla Komal Kalyan/Advanced/NewsFeed/app/src/main/java/com/example/newsfeed/FeedDetails.java

```diff
@@ -0,0 +1,132 @@
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
+import com.bumptech.glide.request.RequestOptions;
+import com.google.android.material.appbar.AppBarLayout;
+import com.google.android.material.appbar.CollapsingToolbarLayout;
+
+public class FeedDetails extends AppCompatActivity implements AppBarLayout.OnOffsetChangedListener{
+
+    private ImageView imageView;
+    private TextView appbar_title, appbar_subtitle, date, time, title;
+    private boolean isHideToolbarView = false;
+    private FrameLayout date_behavior;
+    private LinearLayout titleAppbar;
+    private AppBarLayout appBarLayout;
+    private Toolbar toolbar;
+    private String mUrl, mImg, mTitle, mDate, mSource, mAuthor;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_feed_details);
+
+        toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle("");
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        final CollapsingToolbarLayout collapsingToolbarLayout = findViewById(R.id.collapsing_toolbar);
+        collapsingToolbarLayout.setTitle("");
+
+        appBarLayout = findViewById(R.id.appbar);
+        appBarLayout.addOnOffsetChangedListener(this);
+        date_behavior = findViewById(R.id.date_behavior);
+        titleAppbar = findViewById(R.id.title_appbar);
+        imageView = findViewById(R.id.backdrop);
+        appbar_title = findViewById(R.id.title_on_appbar);
+        appbar_subtitle = findViewById(R.id.subtitle_on_appbar);
+        date = findViewById(R.id.date);
+        time = findViewById(R.id.time);
+        title = findViewById(R.id.title);
+
+        Intent intent = getIntent();
+        mUrl = intent.getStringExtra("url");
+        mImg = intent.getStringExtra("img");
+        mTitle = intent.getStringExtra("title");
+        mDate = intent.getStringExtra("date");
+        mSource = intent.getStringExtra("source");
+        mAuthor = intent.getStringExtra("author");
+
+        RequestOptions requestOptions = new RequestOptions();
+
+        Glide.with(this)
+                .load(mImg)
+                .apply(requestOptions)
+                .transition(DrawableTransitionOptions.withCrossFade())
+                .into(imageView);
+
+        appbar_title.setText(mSource);
+        appbar_subtitle.setText(mUrl);
+        date.setText(Utils.DateFormat(mDate));
+        title.setText(mTitle);
+
+        String author;
+        if (mAuthor != null) {
+            author = " \u2219 " + mAuthor;
+        } else {
+            author = "";
+        }
+
+        time.setText(mSource + author + " \u2219 " + Utils.DateToTimeFormat(mDate));
```

Comment:
Call requires API level 24 (current min is 16): DateToTimeFormat.

Reply:
Done